app/src/main/res/values/userpreferences.xml

```diff
@@ -17,6 +17,10 @@
   <string name="userprefs_submission_thumbnail_summary_on">Show if available</string>
   <string name="userprefs_submission_thumbnail_summary_off">Always hidden</string>
 
+  <string name="userprefs_submission_thumbnails_position">Thumbnails position (requires restart)</string>
```

Comment:
It should be somehow possible to make this preference to not require restart. I don't have ideas right now but I'll try to check later when I'll have more time.

Reply:
This bothers me too. It seems like the render is on cache or something like that. With thumbnails on the left side and without a restart you'll get a mix between the 2 modes. 

Maybe creating one layout to each each mode is not a workaround. Either way, I did PR it for you to take a look.

Reply:
Well, it's caused by ViewHolder being already created with the layout before the setting change. When you close settings and return to main list, rRecyclerView doesn't clear ViewHolders it already created so old layout is still being used. 
One fix I used in other app is to use different view type for different layouts but I do not like this solution as it adds unnecessary complexity. There must be a better way to do it.

Reply:
OK. So that's why you get both layouts till you restart the app. Let's put on hold this so and try a different approach.

Reply:
Sounds like `SubmissionItemDiffer` is unable to identify this change. Adding a value that indicates the thumbnail's position in `SubredditSubmission` should work.

Reply:
Right, the UI constructor also needs to listen to this new preference.

Reply:
Fixed, just title of the settings needs update.